### ISSUE TITLE
[P2 conditional] Learnable skip-gate in U-Net Up blocks — fallback if sketch leak persists after full training

### ISSUE THREAD

**Issue by AcarBasaran** [ahmetnecirvandogan/sketch-generator#56]
**Priority:** P2 (conditional — only if symptom persists)
**Activation trigger:** full-run smoke outputs (10 epochs × 2,424 samples) still show sketch outlines in predicted `albedo.png` / `roughness.png`
**Related:** #45 (cross-attention text injection — addresses the same root cause from a different angle)

## The symptom

In the 2-epoch smoke run, predicted `albedo` and `roughness` came out as flat gray + visible sketch outlines on top — exactly the input sketch pattern was bleeding through into the output maps. Reference smoke result attached in the meeting notes (cardigan sample, long_sleeve_upper id=1).

## Diagnosis

`pbr_model/model.py` uses a standard U-Net with skip connections at every decoder level:

```python
# Up block forward (pbr_model/model.py)
x = torch.cat([x, skip], dim=1)   # decoder feature + encoder skip
x = self.conv(x)                  # DoubleConv to mix
```

At the **highest-resolution** decoder level the skip is `x0` — the encoder's first layer output. The encoder's first conv only needs a few training steps to become a near-identity edge detector for sketches (sketches *are* edges). So `x0 ≈ sketch_image`.

With limited training, the final DoubleConv hasn't learned weights to suppress this skip content. Result: `x0` passes through to the output ↦ predicted maps look like the sketch on a gray background.

This is canonical U-Net early-training behavior, **not a code bug**. Expected to resolve with more epochs. This issue tracks the architecture-level mitigation if it does not.

## What to do (only if triggered)

Add a **learnable scalar gate** per skip connection. One scalar parameter per `Up` block, initialized at 1.0, gradient-trained alongside everything else. Network learns to lower the gate (toward 0) if the skip is hurting prediction.

```python
class Up(nn.Module):
    def __init__(self, in_c: int, skip_c: int, out_c: int) -> None:
        super().__init__()
        self.up = nn.ConvTranspose2d(in_c, in_c // 2, kernel_size=2, stride=2)
        self.conv = DoubleConv(in_c // 2 + skip_c, out_c)
        # NEW: learnable gate scalar, initialized at 1.0 (preserves current behavior at start of training)
        self.skip_gate = nn.Parameter(torch.tensor(1.0))

    def forward(self, x: torch.Tensor, skip: torch.Tensor) -> torch.Tensor:
        x = self.up(x)
        if x.shape[-2:] != skip.shape[-2:]:
            x = F.interpolate(x, size=skip.shape[-2:], mode="bilinear", align_corners=False)
        # NEW: apply learnable gate to the skip path
        x = torch.cat([x, skip * self.skip_gate], dim=1)
        x = self.conv(x)
        return x
```

After training, inspect `model.up1.skip_gate`, `model.up2.skip_gate`, etc. If the last-stage gate (`up4`) is far below 1.0, the network learned to suppress the input-sketch leak.

## Cost

~10 lines in `pbr_model/model.py`. Backwards compatible — existing checkpoints will need their gates initialized at 1.0 (mismatch on `load_state_dict` would need `strict=False` or a small migration).

## Acceptance criteria

- [ ] Each of the 4 `Up` blocks has a `skip_gate` `nn.Parameter`
- [ ] Initialized at 1.0 so initial training behavior is identical to current model
- [ ] Smoke test (10 epochs × 100 samples) shows decreasing loss
- [ ] Inspect trained gates — `up4.skip_gate` (the last-stage one tied to `x0`) should drift below 1.0 if the bug was real
- [ ] Predicted `albedo` / `roughness` no longer show sketch-shaped outlines after full training

## Alternatives considered

- **More training** — first-line mitigation. Cheap, no code change. Likely resolves the symptom on its own.
- **#45 cross-attention** — gives decoder a stronger context signal so it has something better to substitute for the skip. Bigger change, also addresses other prompt-conditioning weaknesses.
- **Skip dropout** — randomly drop entire skip connections during training. Heavier-handed regularizer; useful as a last resort.

This issue is the *minimum-effort* architecture fix: smallest code change, smallest risk, easiest to revert.

## Estimate

~30 minutes including a smoke verification training run.
